ethereum/contracts/ETHApp.sol

```diff
@@ -36,46 +32,28 @@ contract ETHApp is Application {
         // Increment global nonce
         nonce = nonce.add(1);
 
-        bytes memory data = encodeSendData(msg.sender, _recipient, address(0), msg.value, nonce);
-        emit AppEvent(uint(AppEventTags.SendETH), data);
-    }
-
-    function encodeSendData(
-        address _sender,
-        bytes32 _recipient,
-        address _tokenAddr,
-        uint256 _amount,
-        uint256 _nonce
-    )
-        internal
-        pure
-        returns(bytes memory)
-    {
-        return abi.encode(_sender, _recipient, _tokenAddr, _amount, _nonce);
+        emit Transfer(msg.sender, _recipient, msg.value);
     }
 
-    function handle(bytes memory _message)
+    function submit(bytes memory _data)
```

Comment:
as per my previous comment, this should be called `handle`.

Reply:
https://github.com/Snowfork/polkadot-ethereum/pull/75/commits/3ce50f6763d7cd67b4a9968a76b0de19118fad56